### PR TITLE
Add Gherkin support to CPD

### PR DESCRIPTION
**Pull request by ASBrouwers**
## Describe the PR

This adds support for the Gherkin language to CPD.

https://cucumber.io/docs/gherkin/reference/

## Ready?

- [x] Added unit tests for fixed bug/feature
- [x] Passing all unit tests
- [x] Complete build `./mvnw clean verify` passes (checked automatically by github actions)
- [ ] Added (in-code) documentation (if needed)

### CODE REVIEW

File: pmd-gherkin/src/main/antlr4/net/sourceforge/pmd/lang/gherkin/antlr4/Gherkin.g4 [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+DOCSTRING1
+    : '"""' .*? '"""' ;
+
+DOCSTRING2
+    : '```' .*? '```' ;
+
+// Instructions
+BACKGROUND: 'Background:' ;
+EXAMPLES: ('Examples:' | 'Scenarios:') ;
+FEATURE: 'Feature:';
```
Comment (adangel) [inline pmd-gherkin/src/main/antlr4/net/sourceforge/pmd/lang/gherkin/antlr4/Gherkin.g4:92]: We probably should mention somewhere, that only English is supported - non English variants of Gherkin can't be analyzed by pmd (see https://cucumber.io/docs/gherkin/reference/#spoken-languages)

**Review by adangel — CHANGES_REQUESTED** [review]
Thanks for the PR!

Please have a look at my comments.
We have scheduled a new release (6.47.0) this Saturday already - so I'm not sure whether it will land in 6.47.0 or whether it has to wait for 6.48.0.

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to master:<br />
This changeset changes 0 violations,<br />
introduces 0 new violations, 0 new errors and 0 new configuration errors,<br />
removes 0 violations, 0 errors and 0 configuration errors.<br />
<a href="https://chunk.io/pmd/1f9d679bea3f4148b2d10c2925529251/diff1/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by adangel — APPROVED** [review]
Thanks, looks good!

**Pull request merged**